File: java/src/main/java/mchehab/com/java/MainActivity.java
package mchehab.com.java;

import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import java.util.Arrays;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        SensorsFragment sensorsFragment = new SensorsFragment();
        AccelerometerDataFragment accelerometerDataFragment = new AccelerometerDataFragment();

        List<Fragment> listFragments = Arrays.asList(sensorsFragment, accelerometerDataFragment);
        List<String> listTitles = Arrays.asList("Sensors", "Accelerometer Data");

        ViewPager viewPager = findViewById(R.id.viewPager);
        ViewPagerAdapter adapter = new ViewPagerAdapter(getSupportFragmentManager(),
                listFragments, listTitles);
        viewPager.setAdapter(adapter);

        TabLayout tabLayout = findViewById(R.id.tabLayout);
        tabLayout.setupWithViewPager(viewPager);
    }
}

File: app/src/main/java/mchehab/com/readsensordata/MainActivity.kt
package mchehab.com.readsensordata

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(){

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val sensorsFragment = SensorsFragment()
        val sensorDataFragment = AccelerometerDataFragment()
        val listFragments = listOf(sensorsFragment, sensorDataFragment)
        val listTitles = listOf("Sensors", "Accelerometer Data")
        val viewPagerAdapter = ViewPagerAdapter(supportFragmentManager, listFragments, listTitles)
        viewPager.adapter = viewPagerAdapter
        tabLayout.setupWithViewPager(viewPager)
    }
}